src/engine.ts

```diff
@@ -12,11 +12,20 @@ import {
   Planet,
   Round,
   Booster,
-  Resource
+  Resource,
+  TechTile,
+  TechTilePos,
+  AdvTechTile,
+  AdvTechTilePos,
+  Federation
 } from './enums';
 import { CubeCoordinates } from 'hexagrid';
+import Event from './events';
+import techs from './tiles/techs';
+import researchTracks from './research-tracks'
 
 const ISOLATED_DISTANCE = 3;
+const LAST_RESEARCH_TILE = 5;
```

Comment:
You can use `researchTracks.lastTile(field)` instead

Reply:
see above